### ISSUE TITLE
Improvements to Results page

### ISSUE THREAD

**Issue by KristinaGadzhieva** [FreeUKGen/FreeBMD2#609]
In addition to dependent stories, I noticed other inconsistencies on Beta: 
- [ ] ~1. Table header: the font color in the table header is now magenta instead of bold black.~
- [x] 2.  Number of results **disappeared** (story #575 was closed)
- [ ] ~3.  Filter by, Sort by options **disappeared** (story #572 was closed)~
- [x] 4.  Make all links displayed as 'links' and not underlined + add icons
- [x] 5.  Record types: birthS, marriageS. Make singular.  
- [x] 6. If there is no data for extra information (Mother's Maiden Name, Spouse's Surname etc) 'No data' could be indicated  (related story #604) 
- [x] 7. Transfer a link to 'Help with search results' below the table on Beta (implemented on Test 3)
- [ ] ~8. Suggestion: First Name Surname column could be changed to 'Individual's Full Name'~ 
- [x] 9. To enforce mixed case for first names and upper case for surnames.
- [x] 10. Add a short note for users under the table about 'No data' and dash (‘-’) cases.
- [x] 11. Invent icons for the three left-most links in the bar above the table to pick them out.

**Comment by KristinaGadzhieva** [comment]
@richardofsussex @DeniseColbert @nurifreembd Please have a look at # 4, 5, 6, 8 in the story description.

**Comment by Vino-S** [comment]
_Number of results disappeared (story #575 was closed)_
I'm pretty sure the code was up there around the last sprint. Since there were no recent deployments, I am unsure when it decided to go to town. 
Anyway, It's back up now with improvements on the name display.

 _Filter by, Sort by options disappeared (story #572 was closed)_
I think we removed both options after we changed the results table layout. Please correct me if I am wrong.

**Comment by richardofsussex** [comment]
#572 - yes you are right. Sorting of results is now achieved by clicking on column headings, which is a widely understood and frequently-implemented way of doing this. So the 'sort by' options control is redundant. Similarly, the 'filter by' control is simply replicating what the user can do very simply, by modifying their search. There is no need for it in the results page.

**Comment by richardofsussex** [comment]
Regarding suggestion 8: if you hover over "First name" and "Surname" you will see that each is a separate link. This allows sorting of the results by either first name or surname. The proposed change would remove this facility (and open up the question of _how_ names should be sorted!).

**Comment by richardofsussex** [comment]
Suggestion 5: let's create a story for that. Ironically, the plural form would make sense on the search page, since you may find lots of each event type. Once you are in the results page, looking at one record in the table of results, it is definitely a singular instance of that event. Also, I wonder if having them in upper case is a bit shouty - why not just (e.g.) 'Birth'?

**Comment by richardofsussex** [comment]
Suggestion 7: I implemented this as part of the work on context-sensitive help. @Vino-S, this was in branch fb_188_rbl - please merge this into beta.

**Comment by Vino-S** [comment]
fb_188_rbl is now on beta

**Comment by nurifreembd** [comment]
re 5,  consider lower case/capitals. Be consistent with the rest of the labels on the page. Also consider plurals on the page title please.

**Comment by KristinaGadzhieva** [comment]
3 - The sorting improvements are very useful. However, the sorting option may not be obvious to users at first glance. Can we consider adding a note before the table for users? 
4 - Have we agreed that buttons (links) will look like links?  Personally, I would prefer 'buttons look'.  However, it will be a consistent approach across the three websites, so we need to confirm. 
5 - I think the singular form is acceptable and consistent with other names in the table heading. I agree with the lowercase and capitals, 'Birth' etc. Can we make the change based on this story? Since it is a small change, do we really need a separate story on it ? 
9 - Can we consider making the lowercase and capital consistent for all cells ?

**Comment by Vino-S** [comment]
_4 - Have we agreed that buttons (links) will look like links? Personally, I would prefer 'buttons look'. However, it will be a consistent approach across the three websites, so we need to confirm._
Available in test2
<img width="1834" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBK29XQ0E9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--d66e41fdabce6c9d15b594e871f37ee36015446d/image.png" alt="image.png" />


 _5 - I think the singular form is acceptable and consistent with other names in the table heading. I agree with the lowercase and capitals, 'Birth' etc. Can we make the change based on this story? Since it is a small change, do we really need a separate story on it ?_
Ready  for testing in test3
<img width="1751" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBK2NXQ0E9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--20bc25e2de9dbc40205e1007cc1284c0bf75f926/image.png" alt="image.png" />

**Comment by KristinaGadzhieva** [comment]
_5 - I think the singular form is acceptable and consistent with other names in the table heading. I agree with the lowercase and capitals, 'Birth' etc._ 
_9 - Can we consider making the lowercase and capital consistent for all cells ?_

Thanks, Vino.
How about to make First name and Surname make also consistent, with the lowercase and capitals. @richardofsussex @DeniseColbert , do you agree?

**Comment by KristinaGadzhieva** [comment]
4 - Agreed: to make all links displayed as 'links' and single underlined. @Vino-S could you please make changes on Test 3

**Comment by KristinaGadzhieva** [comment]
_3 - The sorting improvements are very useful. However, the sorting option may not be obvious to users at first glance. Can we consider adding a note before the table for users?_

@richardofsussex @DeniseColbert @nurifreembd do you agree ?

**Comment by nurifreembd** [comment]
Would work Khristina. Another option is if user hoovers over the label
maybe  an alt text on the label, would that work?

**Comment by richardofsussex** [comment]
If we're flagging that column headings are links which will cause the table of results to be re-sorted, we should do so in an accessible manner. I don't think alt text (on its own) meets that requirement. Just checked FreeReg, and the headings in their results listing are 'naked' - there is nothing at all to indicate that they trigger a sort. So we're breaking new ground here.

More generally, do we want to implement a way of telling the user what will happen/where they will end up, when they click on a link? For example, the links in the first column lead to the Entry information page for that record - there is nothing to indicate what will happen when you click on the link. (There used to be, when it was a button. Just saying ...) @AlOneill isn't there an 'accessibility' attribute that we could use for this?

**Comment by richardofsussex** [comment]
As regards capitalization of names - I wonder if we would be sacrificing fidelity to the source data in the pursuit of visual tidiness. Names are recorded in the index using a variety of case strategies: shouldn't we simply reflect what was recorded? 

If we _are_ going to enforce a case strategy, I would note that it is common practice (e.g. in FreeReg and the Probate books I am looking at) to have forenames in mixed case and surnames in upper case. @AlOneill does all upper case present itself as "shouting" in a screen reader, or is it not an issue?

**Comment by AlOneill** [comment]
@richardofsussex 
A screen reader user will not know that words are in upper case.

As mentioned in #606, please see https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-sort for details of how to implement the sort accessibly. Note that this is a case for buttons because the sort does something rather than going somewhere.

**Comment by richardofsussex** [comment]
I have implemented support for sorting the table along the lines suggested by @AlOneill.  This includes using buttons and including appropriate ARIA attributes. It's now on test3 for us to look at.

**Comment by KristinaGadzhieva** [comment]
As for capital letters in names (9), consistency in case looks neater. If you look at other competitor genealogy websites, many of them use capital and lowercase letters in names. If we can implement this without modifying the database, I think it can be done in MVP.

**Comment by KristinaGadzhieva** [comment]
@richardofsussex @AlOneill As for sorting (3), sorry, I do not support placing buttons inside the table headers, as it makes the table visually cluttered. We have alternatives, such as making the headings clickable with small sorting icons (up/down arrows) beside the column names. Another option is a drop-down list for sorting, which we had previously implemented (before it was removed).

**Comment by richardofsussex** [comment]
I used W3C guidance as the basis for my work: see https://www.w3.org/WAI/ARIA/apg/patterns/table/examples/sortable-table/. Their advice suggests that it makes sense to give users as big a target to aim at as possible. And as @AlOneill has pointed out, clicking on the heading leads to a 'button-like' operation, not a 'link-like' one.

The W3C example styles the buttons so that they fill the whole heading cell, which gives a neater visual effect than what I managed to achieve using our standard CSS button styles. We could look into doing something similar.

What I have written gives us, as far as I understand it, properly accessible control over table sorting. I would be reluctant to give that up.

**Comment by richardofsussex** [comment]
As regards capitalization, we have established that all caps would not be an issue for screen readers. I don't have strong views on this, and I think it would be easy to implement consistent upper and lower case. Any other opinions?

**Comment by AlOneill** [comment]
@KristinaGadzhieva @richardofsussex 
The sort mechanism should be marked up as an **HTML button** (so that it behaves in the expected way for a keyboard-only user) and present a recognisable interface (such as the back-to-back triangles). It does not need to look like,say, our Search button, because the triangles explain the function. 

@nurifreebmd
An explanation obtained by hovering is inaccessible to keyboard-only users, so let's not go there.

**Comment by nurifreembd** [comment]
Kristina,

can you reconsider attached, think its alt text when you hover over the
column menu on gmail: when user hover over the table column a message
should appear showing they can sort etc.

**Comment by nurifreembd** [comment]
forgot to mention:
 when user tabs through the table columns,
on tabbing onto the column
the code is to be written to show the text that user can filter that column.

**Comment by richardofsussex** [comment]
Ok, this is where I have got to. The source HTML for the table headings looks like this:

![image](https://github.com/user-attachments/assets/4b08f457-1e12-46e2-9e17-9fdbebf09fb3)

and this yields the following result:

![image](https://github.com/user-attachments/assets/35eb4b36-a216-4992-bcf9-26bcf9142aba)

**Comment by nurifreembd** [comment]
Richard,

Is it a button filter?


Ok, this is where I have got to. The source HTML for the table headings

**Comment by richardofsussex** [comment]
OK, I have improved the HTML markup so that the sortable headings are encoded as `<button>` elements. They are styled using our CSS for buttons, and are wrapped in `<a>` elements which carry out the sorting. They have 'sortable' markers:

![image](https://github.com/user-attachments/assets/d99b2988-65e2-4b69-adb0-0691ab4b2731)

**Comment by nurifreembd** [comment]
Why are we agreeing to a button to filter table heading?

Filter table heading is not a button, a button is used to submit a form. An
up and down arrow next to the table header is a standard approach.

Can we agree to Alison’s suggestion too before making suggestions?

**Comment by richardofsussex** [comment]
The result looks similar (apart from the 'sortable' triangles):
![image](https://github.com/user-attachments/assets/a4dab541-eb8f-4108-8dc4-6e5496f3bb1a)

**Comment by richardofsussex** [comment]
I'm happy to include 'Reference' (or '& Reference') in the Registration District heading to make it a bit neater visually. Thoughts?

**Comment by nurifreembd** [comment]
It looks like a button and it is a button style used on sister websites.
Screen reader will read it out as a button with no role.

Kristina, do you agree to this design?

We need to create a stencil where standard objects like a button, filter
etc is defined and is not used inappropriately.

**Comment by DeniseColbert** [comment]
I really dislike the visual of these buttons inside a field/box. How would it be if the column headers were _just_ buttons, rather than bounded by borders?

**Comment by richardofsussex** [comment]
If I take out the CSS styling we get this:

![image](https://github.com/user-attachments/assets/c6aef026-be5c-4385-97c4-b495aa39b697)

Is that what you meant? An alternative strategy is to style the headings so they fill the cell completely (though I'm not sure what would happen with the first column, which contains two sortable headings).

**Comment by nurifreembd** [comment]
Attached is a standard table filter column. Consider applying this standard
as it’s well knowing standard.

**Comment by DeniseColbert** [comment]
Sorry, I didn't explain that very well! Here's a quick mock up of what I mean:

![Untitled drawing (2)](https://github.com/user-attachments/assets/6614b345-7f16-4f2d-bce6-764ccf1b605c)

We'd need to have the buttons well aligned, and "Reference" does look quite odd... I also think the solid magenta is a bit much. Maybe white text and fill and magenta border on the buttons? Or grey like your previous snip?

**Comment by Vino-S** [comment]
Thanks Denise,
Just thinking: Should we go back to when the Sort option  was a separate drop-down. It will remove the link from the table header and also reduce the clutter.

**Comment by richardofsussex** [comment]
The W3C example looks like this:

![image](https://github.com/user-attachments/assets/7cf89f96-1436-47cd-a07a-189db13618f0)

I think it's important that the column headings look like column headings, i.e. they are part of the table, not floating above it.

**Comment by nurifreembd** [comment]
Looks good, any reason why the column is not filled in completely, leaving
say half a cm white border?

Re reference, if it is not sortable then it should be inside and not left
outside. Other option is to have a separate column.

**Comment by richardofsussex** [comment]
@AlOneill thinks that having the sorting controls separate from the table gets a big thumbs down from an accessibility point of view. I'm trying to embed that functionality into the table in a W3C-approved manner.

**Comment by AlOneill** [comment]
@richardofsussex
You are doing well! 
Just a quick reminder that W3C sometimes forget that screen reader users are not the only group that needs consideration wrt accessibility. Use what you know about others such as keyboard-only users to check your work.

**Comment by nurifreembd** [comment]
Richard,

Can we think about the previous design per Vino’s suggestion.

The other standard option is the standard table folder ensuring name, role
description is added to the underlying code.

The design needs to be worked on proposed by Denise as it looks floating
but each column needs to look as part of the table design.

**Comment by Vino-S** [comment]
Thanks Richard for info on separate dropdown.
Is it possible to have a different style for the buttons within the table?
Something like the style used in w3c table you shared.
 Or like style used in UK Gov website
![image](https://github.com/user-attachments/assets/e574362e-bf70-41a3-9b01-2264d39438dd)

**Comment by nurifreembd** [comment]
Attached improved Design by Denise.

Headers is still part of the table design and I think the column should be
filled in more so it does not look like existing buttons.

When user tab through these filter, add show a descriptive text that user
can filter.

These are standard approach.

**Comment by nurifreembd** [comment]
Vino

That is what am trying to explain, exactly the same design. Problem is
because the colour it looks like a button.

Can we completely fill it or leave it white or grey?

Thanks

**Comment by nurifreembd** [comment]
Attached design will suit all users providing we show on tab text to filter
and assign role. Name and description in the code.

Think I have stated the standard approach and options.

Thank you

**Comment by richardofsussex** [comment]
I'm sure we can do better with the style of the buttons. I'll have a think.

**Comment by nurifreembd** [comment]
Richard,

if you do apply the up/down arrows
when user hover over up arrow with a mouse
then show text label up
else when user hover over down arrow with a mouse
then show text label down

or
if user tabs onto the column using a keyboard
then show text label stating 'filter Name and surname by using up and down
arrow'  (something along this line)
we need to consider how the underlying code can be written for user to
toggle up and down arrow using the up and and down on keyboard before going
onto the next column

screen readers
name role and value

Name: column filter up and down arrow
role toggle up and down in alphabetical order
value : values filtered and presented in order up or down

**Comment by richardofsussex** [comment]
Not perfect, but if we drop the option to sort by forenames, and include '& reference' in the District Name heading, the headings do line up:

![image](https://github.com/user-attachments/assets/590cb5ec-e754-44ef-be21-1070ab99ad09)

**Comment by richardofsussex** [comment]
Is that good enough for initial testing?

One detail I have yet to sort out is to determine whether the sorted column is in ascending or descending order. This affects both the ARIA attribute value and the icon that is displayed.

**Comment by nurifreembd** [comment]
Small suggestion attached. Option to Fill in blank areas of the cell or
reduce the height. Just my thoughts.

**Comment by AlOneill** [comment]
@richardofsussex
Getting there! 
However, I'm a bit dubious about "buttons wrapped in links that do the actual sorting". I'll need to test with VO to know whether or not that behaves as intended.

**Comment by richardofsussex** [comment]
I went searching for wisdom on the web and found e.g. https://www.shecodes.io/athena/17038-how-to-add-a-link-to-my-button-in-html. So I followed that pattern. `<button>` has a URL attribute, but only for 'submit' buttons (i.e. form submission). If you can suggest a better encoding strategy I will look into persuading Rails to deliver it.

**Comment by AlOneill** [comment]
@richardofsussex 
Have you tried to activate the button from the keyboard? Does it respond to both Enter and Space? Can you bring the button into focus with Tab? Those would be the expected behaviours. We will find out in due course, on a live example, whether VO lists them as links or buttons.

**Comment by richardofsussex** [comment]
It works as you describe for Tab and Enter. Space causes the cursor to scroll down the table.

**Comment by nurifreembd** [comment]
To reiterate buttons should not be used for table column filter. It will
fail in audit.

**Comment by richardofsussex** [comment]
I've solved the other problem, which is adjusting the display for ascending and descending order, so I've asked @Vino-S to update test3 so you can all give it a spin. (Off to London now for a birthday meal with our son - don't expect anything else today!)

**Comment by Vino-S** [comment]
test3 is updated with the changes

**Comment by nurifreembd** [comment]
Thank you Vino.

Attached on iPhone 13.

Looks like buttons above.

Please consider filling in the cells with the black lined borders.

**Comment by AlOneill** [comment]
@richardofsussex @Vino-S 
The behaviour in response to a pointing device is as expected.

Using only the keyboard, after a sort the focus goes to the very top of the page. From there many presses of Tab are required to return to the table headers. In other words, the behaviour is broken.

The behaviour in VO is equally, but differently, disordered. Also broken.

It looks as if the presence of the link is the cause of all the broken disorder. 

Why do we need the link? Cannot the button perform all that is necessary?

FWIW, the sort buttons in Wikipedia tables do not have an associated link and behave as expected in all three situations: pointing device, keyboard and VO. In all 3 cases, the sort is performed without moving the focus (no link). I'm not saying that Wikipedia presents a model of accessible behaviour, far from it, but it does get some things right.

**Comment by richardofsussex** [comment]
I have just checked the W3C example. The HTML markup is simple, with `<button>` elements and, as you say, no links. However, they are not magic buttons. That example has a raft of javascript code which actually does the sorting. Our sorting is not done that way: we refresh the page with the table differently sorted. I have no idea how we would do the sorting using javascript, and I suspect that it isn't possible. (@Vino-S will know better than I ...) Obviously, by actioning a link which refreshes the page, we cause the cursor to be back at the top of the page.

**Comment by AlOneill** [comment]
@richardofsussex 
In other words, **our current way of sorting will remain inaccessible to keyboard-only users**, both sighted and unsighted. This is disappointing to say the least. In fact I think it will require approval higher up as this broken behaviour goes against the FUG aim of making the apps accessible to all users.

I take back all I said about the sort needing to be activated by buttons. It remains a button-like activity, but as the control is via links, then that is how it should look and behave. We will also need to update the Accessibility Statement with a big apology to affected users — keyboard-only and screen reader users.

I note that Wikipedia uses JavaScript. It works quickly and unobtrusively.

**Comment by richardofsussex** [comment]
This sorting behaviour is no different from what FreeREG and FreeCEN do. (Though they do have links, not buttons. That is easily implemented.) I disagree that the sorting mechanism is inaccessible to keyboard-only users. The fact that the focus does not remain on the column you have just asked to sort is inconvenient, but doesn't make sorting impossible. If you want a second sort, you just have to tab through a few more controls to get there.

It has the approved markers for sortable columns, and implements ARIA support to indicate which column is currently sorted (and in which direction). This is more than the other two sites have achieved. I see no need either for apologies to users or for reference to the Trustees.

**Comment by AlOneill** [comment]
It does not surprise me that REG and CEN have the same issues.

There is a big space between impossible and convenient: it takes **16 presses of Tab** to get back to the beginning of the header row. For someone with cerebral palsy, say, that represents a significant expenditure of energy and concentration. It is likely to be more than 'inconvenient' and  probably not something that such a user would want to repeat. This is not the expected behaviour.

For a screen reader user, I would have to test again once the links are restored — the present behaviour is unhelpful — I'm not hearing anything about the sorting direction or even that the column has been sorted. I hope the current behaviour is down to confusion (on the part of the screen reader software) as to whether the control is a link or a button.

**Comment by AlOneill** [comment]
I tried using a screen reader on REG. The sort works but the user is taken back to the very top of the page. As you say, there is no ARIA to help a user understand what has happened.

For a sighted keyboard-only user, it takes **15 presses of Tab** to get back to the beginning of the header row.

I would say that both sites are broken in this respect and that the results tables are pretty much inaccessible: "not impossible" is hardly good enough.

**Comment by KristinaGadzhieva** [comment]
Hi everyone, I read all the comments above and also did some research.
**Sorting.** It seems that without Javascript, it will be challenging to dynamically manage the sorting, and we would have accessibility issues.  @Vino-S , could you please investigate how long it would take to implement sorting using Javascript. It is certainly possible, but it can take longer. 
If we conclude that Javascript implementation will take too much time, we need to consider compromise options for the MVP that minimize accessibility issues:

- Table-based sorting: 
As I’ve mentioned above, I cannot support the current ‘buttons view’ inside table headers. Instead, we could consider sorting via links within the table headers as a temporary solution until we achieve W3C example in a future version. We can also provide clear ARIA attributes, logical tab order, and visible sorting indicators (up/down arrows). The current diamond-like symbols as sorting indicators may not be immediately clear to all users. Could we separate these symbols, use upward and downward triangles '▲' '▼' opposite each other?  

Additionally, we can consider using scope="col" for column headers and scope="row" for row headers in the  <th> elements (to define each cell more clearly); consider using ARIA Live Regions to notify users when the table has been updated due to sorting. 

**As other alternatives (temporary solutions):**  
- Filter function. We could consider providing filter function that allows users to filter the table by column or by keywords.  Of course, filtering does not fully replace sorting, but it’s an easier and safer temporary solution that can be accessible for the MVP. 
- Consider making a separate drop-down menu more accessible. In this case, we should compare which sorting method (table-based or separate drop-down menu) options is easier to make more accessible and which would cause fewer accessibility issues.

In conclusion, for the MVP version, the first priority is to have basic functionality that works. The second priority is getting user feedback. In other words, we need a working product (simple but functional) which will need development further. Otherwise, we can go in development circles for a long time. We do need many improvements, and possibly even redesign of Beta, but in future phase. Sorting via Javascript and ensuring full accessibility with keyboard navigation, ARIA live regions, and proper indicators will take time. Taking this into account, it makes sense to choose the simplest solution for sorting, but still usable for most users (for now).

**Comment by richardofsussex** [comment]
I've also been thinking about this issue (while not sleeping). The core of the problem as outlined by @AlOneill appears to me to be the fact that you land at the top of the page after each sort operation. I have now implemented a link-based sort which includes a '#' value at the end of the URL, which takes the user back to the table header cell which activated the sort:

![image](https://github.com/user-attachments/assets/6bdc280a-b52d-43ef-843a-74c256ece4f4)

When @Vino-S has had chance to upload my code to test3, you can try it out. My own view is that this works adequately, and there is no need to switch to a javascript-based approach.

**Comment by richardofsussex** [comment]
OK, it's there now: thanks Vino. 

The table has the aria-sort attribute, and uses aria-hidden on the sort icons as recommended. So I think the HTML markup is as it should be.

I think that putting 'up' and 'down' markers on each sortable column would confuse rather than clarify. since it would make the currently sorted column less obvious.

**Comment by KristinaGadzhieva** [comment]
An arrow icon indicates sorting direction (ascending or descending) in the column header. Unsorted columns display a neutral double-arrow icon.  The double-arrow icon is widely used.  The current diamond-like symbol can be confusing and is not commonly used on websites.  (https://daext.com/wp-content/uploads/2024/10/datatables-table-with-sorting.png)

**Comment by richardofsussex** [comment]
I read this on the W3C ARIA table sorting page (https://www.w3.org/WAI/ARIA/apg/patterns/table/examples/sortable-table/):

<img width="972" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBKzBtQ0E9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--0625eb26a49649ba042dc88cdcad927bad79d342/image.png" alt="image.png" />

and found the symbol the closest I could find to the one they use there.

**Comment by AlOneill** [comment]
Progress: for a sighted keyboard-only user the behaviour is as expected (or close enough)—thank you.

However, for a screen reader user (on VO, at least), the sort takes a user back to the very top. I cannot offer an explanation. I'm not hearing any ARIA-related messages. Can I suggest the caption begins "Select a column heading …" rather than "Click on a …" as the only people to hear the caption won't be able to click?

**Comment by richardofsussex** [comment]
Well ... my first experience of a screen reader (as far as I can recall). I used Narrator to cross-check your findings. 

When I pressed Enter to sort by a column, it did indeed reload the page, and gave some initial description of it, but after that, when I pressed Tab once, the focus moved straight back to the column heading I had sorted on, exactly as it should. Try pressing Tab yourself a few times, to confirm (or, hopefully, not) that you really are at the top of the page! Thanks

Caption amended as suggested. For some reason Narrator reads it out twice.

**Comment by richardofsussex** [comment]
No mention of `aria-sort` in Narrator either, but then Narrator fails the tests in https://a11ysupport.io/tests/html/aria/aria-sort.html so I don't think that's our problem. (You could try it with VO, to see if that software recognizes `aria-sort.)`

**Comment by AlOneill** [comment]
Sorry, but after a sort, VO returns to the top of the page and starts reading. This happens if I use the keys that VO tells me to use or whether I use Enter. Pressing Tab does not help.

No mention of sort or sort-direction in the link you gave in VO either. Have to stop now as getting heart rate warnings.

**Comment by KristinaGadzhieva** [comment]
Thank you all.
Minor things left: 
4. Display all links as 'links' with a single underline. 
9. Ensure consistent use of lowercase and capital letters across all cells (including First Name and Surname).

I have added information about 'No data' and dash (‘-’) cases to the Help MVP doc. 
However, could we also add a short note for users under the table? It's easier to have clarification right under the table rather than searching for it on the Help page.

Regarding sorting, have we agreed on table-based sorting despite some accessibility issues? To what extent are we going to address this in the MVP?

**Comment by richardofsussex** [comment]
Agreed to enforce mixed case for first names and upper case for surnames, noting that this is consistent with practice in FreeCEN and FreeREG, but different from what many competitor sites do. Testing/feedback will reveal (hopefully) what our users' preferences are.

**Comment by KristinaGadzhieva** [comment]
@Vino-S , could you please take **4 and 9?**
**4. Make all links displayed as 'links' and not underlined+add icons for the three left-most links in the bar above the table
9. To enforce mixed case for first names and upper case for surnames.**
 
@richardofsussex **could you please help with 10?** 
 **10. Add a short note for users under the table about 'No data' and dash (‘-’) cases.**  **Alison's comment on this:** "If you are going to explain anything about the table, please add a heading to that explanation so that a screen reader user can discover it (without having to read the whole page)."

**Comment by DeniseColbert** [comment]
4.
@Vino-S Icon suggestions

**View search criteria**
![image](https://github.com/user-attachments/assets/e79c2a61-b707-43a2-b3e2-b7ec2722cbd5)
https://fontawesome.com/search?q=view&o=r&m=free

**Revise search**
![image](https://github.com/user-attachments/assets/d32abf01-3e16-4d1a-a284-bd4e0ba5f822)
https://fontawesome.com/icons/pen-to-square?f=classic&s=regular

**New search**
![image](https://github.com/user-attachments/assets/25508574-c570-4d5e-a84a-654863775292)
https://fontawesome.com/icons/magnifying-glass?f=classic&s=solid

**Comment by nurifreembd** [comment]
All three icons seem reasonable for a standard user.

For screen users please define role, name and description for each icon
please.

**Comment by richardofsussex** [comment]
I have done 10. and it is implemented in test3. It looks to me as though 9. has been implemented too.

**Comment by KristinaGadzhieva** [comment]
9 and 10 look good, thank you.
Only icons are left (11. Invent icons for the three left-most links in the bar above the table to pick them out)

**Comment by richardofsussex** [comment]
Can @Vino-S take on 11 please?

**Comment by Vino-S** [comment]
Deployed to Beta

**Comment by KristinaGadzhieva** [comment]
10 is not on Beta

**Comment by richardofsussex** [comment]
See #604 comment. Just needs pushing to beta.

**Comment by richardofsussex** [comment]
![Image](https://github.com/user-attachments/assets/8460946d-543b-48d7-8fe4-189aef346bfb)

**Comment by Vino-S** [comment]
Deployed
